inference-engine/src/vpu/graph_transformer/src/middleend/passes/annotate_memory_types.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "vpu/middleend/pass_manager.hpp"
+
+namespace vpu {
+
+namespace {
+
+template<class DataObjects>
+std::vector<vpu::MemoryType> gather(const DataObjects& dataObjects) {
+    std::vector<vpu::MemoryType> types;
+    types.reserve(dataObjects.size());
+    std::transform(dataObjects.begin(), dataObjects.end(), std::back_inserter(types), [](const Data& data) { return data->memReqs(); });
+    return types;
+}
+
+class PassImpl final : public Pass {
+public:
+    void run(const Model& model) override {
+        for (const auto& stage : model->getStages()) {
```

Comment:
You can use `PerStagePass`

Reply:
I thought about it, but then I saw it still has `model` in interface, so it seems me as no difference

Reply:
I think I will change it if there would be a need to update PR

Reply:
Actually, I just found out that is impossible to use here since  `PerStagePass`  works only on specific stage types list while my past type agnostic.